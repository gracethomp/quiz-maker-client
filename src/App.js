import { BrowserRouter, Routes, Route } from "react-router-dom";
import QuizMaker from './page/quiz-maker';
import QuizVisitor from './page/quiz-visitor';
import QuizAccess from './page/quiz-access';
import Result from "./page/results";

function App() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/maker" element={<QuizMaker />}/>
        <Route path="/:slug" element={<QuizVisitor slug1='random-general-quiz'/>}/>
        <Route path='/' element={<QuizAccess/>}/>
        <Route path='/results' element={<Result/>}/>
      </Routes>
    </BrowserRouter>
  );
}
// render={({ match }) => <QuizVisitor slug={match.params.slug}
export default App;
