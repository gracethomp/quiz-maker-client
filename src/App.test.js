import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import App from './App';
import Question from './components/visitor/question';

test('renders MyComponent without errors', () => {
  render(<App />);
});

test('matches snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test('answer test', () => {
  const props = {
    question: '???',
    answers: [{
      'id': 1,
      'answer': 'answer1'
    }, {
      'id': 2,
      'answer': 'answer2'
    }],
  };
  render(<Question {...props} />);
});

test('handles button click correctly', () => {
  const { getByText } = render(<App />);
  const buttonElement = getByText('Go');
  fireEvent.click(buttonElement);
  expect(buttonElement).toBeVisible;
});