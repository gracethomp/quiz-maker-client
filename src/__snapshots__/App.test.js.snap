// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="form access"
  >
    <div
      class="header"
    >
      <p
        class="header-text"
      >
        quizmaker.com
      </p>
    </div>
    <div
      class="access-form"
    >
      <input
        class="field quiz-slag"
        placeholder="Put quiz-slug here"
        type="text"
        value=""
      />
      <div
        class="creation-buttons"
      >
        <div
          class="access-button form"
          role="button"
        >
          <span
            class="send-text"
          >
            Go
          </span>
        </div>
      </div>
    </div>
    <div
      class="underline"
    />
    <div
      class="creation-buttons"
    >
      <div
        class="create-quiz-button form"
        role="button"
      >
        <span
          class="send-text"
        >
          Create your own quiz
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
